rasa/core/policies/rule_policy.py

```diff
@@ -739,13 +741,18 @@ def _find_action_from_loop_happy_path(
 
     def _find_action_from_rules(
         self, tracker: DialogueStateTracker, domain: Domain
-    ) -> Tuple[Optional[Text], Optional[Text]]:
+    ) -> Tuple[Optional[Text], Optional[Text], bool]:
         tracker_as_states = self.featurizer.prediction_states([tracker], domain)
         states = tracker_as_states[0]
         current_states = self.format_tracker_states(states)
 
         logger.debug(f"Current tracker state:{current_states}")
 
+        # Tracks if we are returning after an unhappy loop path. If this becomes `True`
+        # the policy returns an event which disables the slot validation for the next
+        # loop execution.
```

Comment:
this comment mixes forms and loops. Maybe smth along the lines:
"For example, if `active_loop` is a form, then slot validation is disabled"

Reply:
👍 
It's easy to mix currently as every loop currently is a form 🙈

Reply:
not really. two stage fallback is not a form